source/adapters/cuda/queue.hpp

```diff
@@ -27,6 +29,10 @@ struct ur_queue_handle_t_ {
 
   std::vector<native_type> ComputeStreams;
   std::vector<native_type> TransferStreams;
+  // Stream used solely when profiling is enabled
+  native_type ProfStream;
+  bool IsProfStreamCreated{false};
```

Comment:
I don't think we need this guy? Since profiling stream is created `if (RequiresTimings)` we can just use that same check whenever we use this guy.

Reply:
This was only to check if we need to [destroy](https://github.com/oneapi-src/unified-runtime/blob/55412d81af29e8b8c5715f4f1dac8f629698a7a8/source/adapters/cuda/queue.cpp#L205) that stream when the queue finished. To use `if (RequiresTimings)` we'd need to destroy it directly from the event.

Reply:
I removed the `bool` guy and changed the condition before `destroy` to:
 ```cpp
 if (hQueue->getHostSubmitTimeStream() != CUstream{0})
```

Reply:
Nice one